File: Problem849.java
class Problem849 {
    public int maxDistToClosest(int[] seats) {
        int curr = 0;
        while(seats[curr] == 0) curr++;
        int maxDist = curr;
        for (int i = curr; i < seats.length; i++){
            if (seats[i] == 1){
                maxDist = Math.max(maxDist, (curr+1)/2);
                curr = 0;
            }
            else curr++;
        }
        maxDist = Math.max(maxDist, curr);
        return maxDist;
    }
}

File: Problem849.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Problem849 {
public:
    int maxDistToClosest(vector<int>& seats) {
        int curr = 0;
        while (seats[curr] == 0) curr++;
        int MAX = curr;
        for (int i = curr; i < seats.size(); i++) (seats[i] == 1) ? MAX = max(MAX, (curr+1)/2), curr = 0 : curr++;
        MAX = max(MAX, curr);
        return MAX;
    }
};